honeybee/cli/edit.py

```diff
@@ -0,0 +1,292 @@
+        parsed_model = Model.from_dict(data)
+
+        # convert the units of the model
+        parsed_model.convert_to_units()
+
+        # write the new model out to the file or stdout
+        output_file.write(json.dumps(parsed_model.to_dict()))
+    except Exception as e:
+        _logger.exception('Model unit conversion failed.\n{}'.format(e))
+        sys.exit(1)
+    else:
+        sys.exit(0)
+
+
+@edit.command('solve-adjacency')
+@click.argument('model-json', type=click.Path(
+    exists=True, file_okay=True, dir_okay=False, resolve_path=True))
+@click.option('--wall/--air-boundary', help='Flag to note whether the '
+              'wall adjacencies should be of the air boundary face type.',
+              default=True, show_default=True)
+@click.option('--surface/--adiabatic', help='Flag to note whether the adjacencies '
+              'should be surface or adiabatic.', default=True, show_default=True)
+@click.option('--output-file', help='Optional file to output the Model JSON string'
+              ' with solved adjacency. By default it will be printed out to stdout',
+              type=click.File('w'), default='-')
+def solve_adjacency(model_json, wall, surface, output_file):
+    """Solve adjacency between Rooms of a Model JSON file.
+    \n
+    Args:
+        model_json: Full path to a Model JSON file.
+    """
+    try:
+        assert os.path.isfile(model_json), 'No JSON file found at {}.'.format(model_json)
+        # serialize the Model to Python
+        with open(model_json) as json_file:
+            data = json.load(json_file)
+        parsed_model = Model.from_dict(data)
+        # check the Model tolerance
+        assert parsed_model.tolerance != 0, \
+            'Model must have a non-zero tolerance to use solve-adjacency.'
+        tol = parsed_model.tolerance
+
+        # solve adjacency
+        adj_info = Room.solve_adjacency(parsed_model.rooms, tol)
+
+        # try to assign the air boundary face type
+        if not wall:
+            for face_pair in adj_info['adjacent_faces']:
+                if isinstance(face_pair[0].type, Wall):
+                    face_pair[0].type = face_types.air_boundary
+                    face_pair[1].type = face_types.air_boundary
+
+        # try to assign the adiabatic boundary condition
+        if not surface and ad_bc:
+            for face_pair in adj_info['adjacent_faces']:
+                face_pair[0].boundary_condition = ad_bc
+                face_pair[1].boundary_condition = ad_bc
+
+        # write the new model out to the file or stdout
+        output_file.write(json.dumps(parsed_model.to_dict()))
+    except Exception as e:
+        _logger.exception('Model solve adjacency failed.\n{}'.format(e))
+        sys.exit(1)
+    else:
+        sys.exit(0)
+
+
+@edit.command('windows-by-ratio')
+@click.argument('model-json', type=click.Path(
+    exists=True, file_okay=True, dir_okay=False, resolve_path=True))
+@click.argument('ratio', type=float)
```

Comment:
Can we pass multiple ratios similar to ratio by orientation?

Reply:
We should definitely support that functionality at some point but I think that it should probably be through a separate command or through some way that doesn't make this command more difficult to use.  And we have to agree on some convention for inputting lists into the commands.  Is a JSON string of a list the way you were thinking of implementing it?

Reply:
You can pass multiple value for an option by using multiple=True.

https://click.palletsprojects.com/en/7.x/options/#multiple-options

I think we can include it in this command.

Reply:
But that is for a `click.option` and not a `click.argument`.  Are you suggesting that we just make the ratio optional with a default?

Also, how will queenbee interact with lists of inputs in this format?

Reply:
You can make ratio an option with a default value.

> Also, how will queenbee interact with lists of inputs in this format?

Queenbee supports ArrayInput which is a list of numbers. Let's keep the commands for honeybee clean and resolve any possible issues on the Queenbee side.

Reply:
I guess it's just that trying to add this list functionality really feels un-clean to me. As does changing some of these arguments into options (especially things like `aperture_height` for which a default value is invariably going to be making a potentially incorrect assumption about the units of the model).  In other words, I feel like I'm only adding this list functionality for the sake of queenbee because I'm assuming that queenbee has some more elegant way of inputting lists.